WeakAuras/Prototypes.lua

```diff
@@ -6602,21 +6602,162 @@ Private.event_prototypes = {
   ["Crowd Controlled"] = {
     type = "status",
     events = {
-      ["unit_events"] = {
-        ["player"] = {"UNIT_AURA"}
+      ["events"] = {
+        --"PLAYER_CONTROL_LOST",
+        --"PLAYER_CONTROL_GAINED",
+        --"LOSS_OF_CONTROL_ADDED",
+        "LOSS_OF_CONTROL_UPDATE"
       }
     },
-    force_events = "UNIT_AURA",
+    force_events = "LOSS_OF_CONTROL_UPDATE",
     name = L["Crowd Controlled"],
+    canHaveDuration = true,
+    statesParameter = "one",
+    init = function(trigger)
+      local ret = [=[
+          local show = false
+          local use_controlType = %s
+          local controlType = %s
+          local inverse = %s
+          local use_interruptSchool = %s
+          local interruptSchool = tonumber(%q)
+          local duration, expirationTime, spellName, icon, spellName, spellId, locType, lockoutSchool, controlText, _
+          for i = 1, C_LossOfControl.GetActiveLossOfControlDataCount() do
+            local data = C_LossOfControl.GetActiveLossOfControlData(i)
+            if data then
+              if (not use_controlType)
+              or (data.locType == "PACIFYSILENCE" and (controlType == "SILENCE" or controlType == "PACIFY"))
+              or (data.locType == "STUN_MECHANIC" and controlType == "STUN")
+              or (data.locType == controlType and (controlType ~= "SCHOOL_INTERRUPT" or ((not use_interruptSchool) or data.lockoutSchool == interruptSchool)))
```

Comment:
Did you check what the lookoutSchool is if a spell with multiple spell schools is interrupted?

E.g. frostfire, chaos bolt or whatever else tehre might be?

Reply:
Hmm, yeah I'll see if I can test this later.

Reply:
I checked: for example, lockoutSchool is [124](https://wow.gamepedia.com/COMBAT_LOG_EVENT#Spell_School) when interrupting [Chaos Bolt](https://www.wowhead.com/spell=116858)

Reply:
Thanks for checking. 
That would mean that they would be locked out of all [Fire, Nature, Frost, Shadow, Arcane], right?
So if I aim to provide a dropdown for the base school lockouts I need to associate the higher level school values with each of the base types. Or, actually looks like I can go via a bitmask.

Reply:
Another example: lockoutSchool is [72](https://wow.gamepedia.com/COMBAT_LOG_EVENT#Spell_School) when interrupting [Full Moon](https://www.wowhead.com/spell=274283)
Yes, all my abilities of nature and arcane school are locked out =(

Reply:
Yup `bit.band(data.lockoutSchool, interruptSchool)` should handle it. I'll test tonight to make sure.